### DISCUSSION TITLE
x-for doesn't work in tauri.js application enviroment.

### DISCUSSION THREAD

**Discussion by grey4owl** [alpinejs/alpine#3530] [7. Bugs (archived)] [answered]
Hi, :smiley: 

i've been using alpine.js for a long time now and I really like it! I recently started using alpine.js in a tauri.js environment as well and haven't encountered a single glitch until today.. I tried using `x-for` but i got the following error `Alpine Expression Error: null is not an object (evaluating 'e._x_dataStack=[t,...$(r||e)]')` with this example: 

````html
<ul x-data="{ colors: ['Red', 'Orange', 'Yellow'] }">
    <template x-for="color in colors">
        <li x-text="color"></li>
    </template>
</ul>
````
If I try this same example in the browser without tauri.js, everything works as it should. It's like it can't find the array of `colors`, I tried different ways to define `colors` but I always get the same error. I hope someone might have a solution for this because I really need it..

**Comment by ekwoka** [comment]
Hmm, that's interesting.

Do you have a reference repo or anything?

**Reply by grey4owl** [reply]
https://github.com/grey4owl/alpine-x-tauri

**Reply by grey4owl** [reply]
https://www.tim-kleyersburg.de/articles/alpine-expression-error-x-datastack/

**Reply by ekwoka** [reply]
I am not having any issues with that code at all.

And looking at the Alpine Source, it should not be possible for anything to be null in that expression.

What platform are you running it on? Like what is the webview Tauri is using?

**Reply by grey4owl** [reply]
I'm on fedora linux and tauri uses webkitgtk, I tested the same code on gnome web browser and it works there but when I run it through tauri it won't..

**Reply by grey4owl** [reply]
I just found out that if i build the whole app with `cargo tauri build`, then it works, but it doesn't work in the dev environment `cargo tauri dev`, so this is probably a tauri side problem. Thank you for your time!

**Comment by xjzh123** [comment, marked as answer]
This is a bug of tauri and will be fixed in future versions of tauri.
https://github.com/tauri-apps/tauri/issues/6494#issue-1630769819
